Cabal/Distribution/Simple/Build.hs

```diff
@@ -188,7 +217,7 @@ buildComponent verbosity numJobs pkg_descr lbi suffixes
         installedPkgInfo = inplaceInstalledPackageInfo pwd distPref pkg_descr
                                                        (AbiHash "") lib' lbi clbi
 
-    registerPackage verbosity (compiler lbi) (withPrograms lbi) HcPkg.MultiInstance
+    registerPackage verbosity (compiler lbi) (withPrograms lbi) HcPkg.NoMultiInstance
```

Comment:
Ooh this needs a comment, and also it probably isn't quite right. When you `cabal build`, the inplace database is blown away prior to building. This means that whenever you register, it's for a distinct component, so we don't want ghc-pkg to blow away any old instances; we want everything!

However, in `--one-shot` mode the situation is different: the inplace database must be preserved over calls to `build --one-shot`, since we might first build a convenience library (and inplace register it) before building the main library. But now, if we rebuild the convenience library, we need to make sure the new installation overwrites the previous one. I guess maybe I failed to test for these cases.